[package] cpprestsdk/2.10.19: Enable transitive dependencies

### Description

When using this package, you have to manually include headers for boost and OpenSSL. This is probably because requirements do not have transitive_headers=True

### Package and Environment Details

* Package Name/Version: **cpprestsdk/2.10.19**
* Operating System+version: **Windows 11**
* Compiler+version: **MSVC 194**
* Docker image: **-**
* Conan version: **conan 2.16.1**
* Python version: **Python 3.13.3**


### Conan profile

[settings]
os=Windows
arch=x86
build_type=Debug
compiler=msvc
compiler.cppstd=14
compiler.runtime=static
compiler.runtime_type=Debug
compiler.version=194
os=Windows


### Steps to reproduce

Use the package in a dummy application, without manually including headers for Boost and OpenSSL.

### Logs

Build just fails, not able to find boost/asio/ssl.hpp

Hello, thanks a lot for the quick response.

Error for OpenSSL:
`include/boost/asio/ssl/detail/openssl_types.hpp(23,10): error C1083: Cannot open include file: 'openssl/conf.h': No such file or directory`

--- a/recipes/cpprestsdk/all/conanfile.py
+++ b/recipes/cpprestsdk/all/conanfile.py
@@ -58,7 +58,7 @@ class CppRestSDKConan(ConanFile):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("boost/1.83.0")
+        self.requires("boost/1.83.0", transitive_headers=True)
         self.requires("openssl/[>=1.1 <4]")
         if self.options.with_compression:
             self.requires("zlib/[>=1.2.11 <2]")

